packages/web/src/pages/trending-page/TrendingPageProvider.tsx

```diff
@@ -0,0 +1,94 @@
+import { TrendingPageContentProps } from './types'
+
+/**
+ * TrendingPageProvider using modern React hooks pattern
+ */
+
+const TrendingPageProvider = ({
+  containerRef,
+  children: Children
+}: TrendingPageOwnProps) => {
+  // Custom hooks for state and actions
+  const trendingPageState = useTrendingPageState()
+  const actions = useTrendingActions()
+  const lineups = useTrendingLineups({
+    trendingPageState,
+    containerRef: containerRef || undefined
+  })
+
+  // URL parameter management
+  useTrendingUrlParams({
+    trendingPageState,
+    setTrendingGenre: actions.setTrendingGenre,
+    setTrendingTimeRange: actions.setTrendingTimeRange,
+    replaceRoute: actions.replaceRoute
+  })
+
+  // Cleanup hook
+  useTrendingPageCleanup({ trendingPageState, actions })
+
+  // Simple prop composition using useMemo for performance
+  const childProps: TrendingPageContentProps = useMemo(
+    () => ({
+      // Static messages
+      trendingTitle: TRENDING_MESSAGES.trendingTitle,
+      pageTitle: TRENDING_MESSAGES.pageTitle,
+      trendingDescription: TRENDING_MESSAGES.trendingDescription,
+
+      // Core state
+      trending: {} as any, // Legacy prop, not used
+      trendingGenre: trendingPageState.trendingGenre,
+      trendingTimeRange: trendingPageState.trendingTimeRange,
+      lastFetchedTrendingGenre: trendingPageState.lastFetchedTrendingGenre,
+      hasAccount: trendingPageState.hasAccount,
+
+      // Lineup data
+      trendingWeek: lineups.trendingWeek,
+      trendingMonth: lineups.trendingMonth,
+      trendingAllTime: lineups.trendingAllTime,
+      getLineupProps: lineups.getLineupProps,
+      getLineupForRange: lineups.getLineupForRange,
+      scrollToTop: lineups.scrollToTop,
+
+      // Actions
+      goToSignUp: actions.goToSignUp,
+      goToGenreSelection: actions.goToGenreSelection,
+      setTrendingGenre: actions.setTrendingGenre,
+      setTrendingTimeRange: actions.setTrendingTimeRange,
+      playTrendingTrack: actions.playTrendingTrack,
+      pauseTrendingTrack: actions.pauseTrendingTrack,
+      refreshTrendingInView: actions.refreshTrendingInView,
+      resetTrendingLineup: actions.resetTrendingLineup,
+
+      // Action factories
+      makeLoadMore: actions.makeLoadMore,
+      makePlayTrack: actions.makePlayTrack,
+      makePauseTrack: actions.makePauseTrack,
+      makeSetInView: actions.makeSetInView,
+      makeRefreshTrendingInView: actions.makeRefreshTrendingInView,
+      makeResetTrending: actions.makeResetTrending,
+
+      // Legacy/unused props
+      goToTrending: () => {}, // Not defined in original
+      switchView: () => {}, // Not defined in original
+      fetchSuggestedFollowUsers: () => {}, // Not used
+      followUsers: () => {}, // Not used
+      suggestedFollows: [], // Not used
+      setTrendingInView: () => {} // Not used
+    }),
+    [trendingPageState, actions, lineups]
+  )
+
+  return <Children {...childProps} />
```

Comment:
So to me the biggest issue with this pattern is this line, I'd like us to get rid of this complexity and have trending page be a non provider page where there relevant complaints implement the logic rather than provider and pro drilling. Potentially it's a responsive page too